pvHelpers/device_certificate.py

```diff
@@ -0,0 +1,45 @@
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.x509.oid import ExtensionOID, NameOID
+import pvHelpers as H
+
+class DeviceCertificate:
+    def __init__(self, cert_string):
+        self.cert_string = cert_string
+
+    def verify_certificate(self):
+        try:
+            # Load the certificate
+            x509.load_pem_x509_certificate(self.cert_string.encode(), default_backend())
+            return True
+        except ValueError:
+            return False
+
+    def get_certificate_details(self):
+        try:
+            cert = x509.load_pem_x509_certificate(self.cert_string.encode(), default_backend())
+
+            # Extract common name from subject and issuer
+            subject_cn = cert.subject.get_attributes_for_oid(NameOID.COMMON_NAME)
+            issuer_cn = cert.issuer.get_attributes_for_oid(NameOID.COMMON_NAME)
+
+            issued_to = subject_cn[0].value if subject_cn else "N/A"
+            issued_by = issuer_cn[0].value if issuer_cn else "N/A"
```

Comment:
Can we be sure that there will always be at least one value in the array?

Reply:
https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Name.get_attributes_for_oid

> If nothing matches an empty list will be returned.

So if it's an empty list then we'll display "N/A". If its populated then we'll show the first value in the list.

Reply:
An empty list will be `[]` and not `None`, so `subject_cn[0]` will throw an exception.

Reply:
Doesn't python treat empty lists as `False` in a boolean context?

`if subject_cn` or `if []` would return false so it'll assign the variable to `"N/A"`

```
x = []

issued_by = x[0].value if x else "N/A"

print(issued_by) # prints N/A

```

Reply:
Right you are! I actually didn't realize that length 0 implied truth in python.